pkg/util/cert/crl.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cert
+
+import (
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"io/ioutil"
+	"sort"
+)
+
+// RevocationPolicy is a strategy for querying CRLs and evaluating if a certificate
+// has been revoked.
+type RevocationPolicy interface {
+	VerifyCertificate(cert *x509.Certificate) (valid bool, err error)
+}
+
+// LoadCRLFile parses a revocation list from a file and uses it to statically mark
+// certificates as valid or revoked. The signatures of the CRL and the CRL distribution
+// points of passed certificates are not evaluated.
+func LoadCRLFile(file string) (RevocationPolicy, error) {
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("read file: %v", err)
+	}
+	certList, err := x509.ParseCRL(data)
+	if err != nil {
+		return nil, fmt.Errorf("parse crl: %v", err)
+	}
+
```

Comment:
Since that generally indicates that the CRL needs to be refreshed, and there's no way to refresh a file based CRL, I omitted it. I could see checking it to ensure your API server always has an up to date list.

Do we want the `--crl-file` flag to fail if the CRL has expired? I guess doing so would bring it in line with other X509 assets e.g. your API server wont start if it has an expired cert.